### PR TITLE
Ensure string is returned when applying filter to email template

### LINKED ISSUE #32671: [Issue] Ensure string is returned when applying filter to email template
This issue is automatically created based on existing pull request: magento/magento2#32298: Ensure string is returned when applying filter to email template

---------

### Description (*)
Added string casting within the email template filter method to ensure the return value is a string.
Currently if an exception is caught while not in developer mode a Phrase object is returned. This in turn triggers a fatal Uncaught TypeError in magento/framework/Filter/DirectiveProcessor/TemplateDirective.php:68.

### Related Pull Requests
N/A

### Fixed Issues (if relevant)
N/A

### Manual testing scenarios (*)
1. Install module
[template_error.zip](https://github.com/magento/magento2/files/6263499/template_error.zip)
2. Switch to Production mode.
3. Go to Marketing -> Newsletter Templates and create a new one.
4. Insert `{{template config_path="design/email/header_template"}}` into Template Content filed.
5. Preview Template.
6. `We're sorry, an error has occurred while generating this content.` message displayed.

### Questions or comments
N/A

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### PR DESCRIPTION
**Pull request by Skullsneeze**
### Description (*)
Added string casting within the email template filter method to ensure the return value is a string.
Currently if an exception is caught while not in developer mode a Phrase object is returned. This in turn triggers a fatal Uncaught TypeError in magento/framework/Filter/DirectiveProcessor/TemplateDirective.php:68.

### Related Pull Requests
N/A

### Fixed Issues (if relevant)
N/A

### Manual testing scenarios (*)
1. Install module
[template_error.zip](https://github.com/magento/magento2/files/6263499/template_error.zip)
2. Switch to Production mode.
3. Go to Marketing -> Newsletter Templates and create a new one.
4. Insert `{{template config_path="design/email/header_template"}}` into Template Content filed.
5. Preview Template.
6. `We're sorry, an error has occurred while generating this content.` message displayed.

### Questions or comments
N/A

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### Resolved issues:
1. [x] resolves magento/magento2#32671: Ensure string is returned when applying filter to email template

### CODE REVIEW

**Comment by Skullsneeze** [comment]
@magento run all tests

**Comment by Skullsneeze** [comment]
@magento run all tests

**Review by ihor-sviziev — APPROVED** [review]
✔ Approved.

Failing tests looks not related to changes from this PR.

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-8835](https://jira.corp.magento.com/browse/ENGCOM-8835) has been created to process this Pull Request

**Comment by magento-engcom-team** [comment]
@Skullsneeze thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

**Comment by engcom-Bravo** [comment]
Dev experience is required for testing this PR. Moving it to the _Extended Testing_ column

**Comment by engcom-Foxtrot** [comment]
Hello, @Skullsneeze! Your fix looks reasonable, but our workflow requires obvious steps to reproduce a fixed issue. I was unable to find any scenario for reproducing your issue.  Please, add a testing scenario to this PR and we should process with this PR.

File: app/code/Magento/Email/Model/Template/Filter.php [hunk 1]
```diff
@@ -1086,7 +1086,7 @@ public function filter($value)
             if ($this->_appState->getMode() == \Magento\Framework\App\State::MODE_DEVELOPER) {
                 $value = sprintf(__('Error filtering template: %s'), $e->getMessage());
             } else {
-                $value = __("We're sorry, an error has occurred while generating this content.");
+                $value = (string) __("We're sorry, an error has occurred while generating this content.");
```
Comment (engcom-Foxtrot) [inline app/code/Magento/Email/Model/Template/Filter.php:1089]: Hello, @Skullsneeze! Your fix looks reasonable, but our workflow requires obvious steps to reproduce a fixed issue. I was unable to find any scenario for reproducing your issue. Please, add a testing scenario to this PR and we should process with this PR.

**Review by engcom-Foxtrot — CHANGES_REQUESTED** [review]

**Comment by ihor-sviziev** [comment]
Hi @Skullsneeze,
Will you be able to provide some detailed info on how to reproduce this issue?
Thank you!

**Comment by engcom-Foxtrot** [comment]
QA passed.
@Skullsneeze @ihor-sviziev steps to reproduce added.

**Comment by engcom-Oscar** [comment]
@magento create issue

**Pull request merged**